Test failure on MacOS Sequoia in numexpr/numpy comparison

Seeing a failure in `holopy/scattering/tests/test_lens.py::TestLens::test_integrand_prefactor_same_with_numexpr_as_without()` on the latest `develop` branch under MacOS 15.2 (Sequoia) with numpy 1.20.3, numexpr 2.10.0.  This failure doesn't occur in our `macos-latest` CI test runner, which uses MacOS 14.7, so it seems to be related to some change in the update to MacOS 15. 

```
_______ TestLens.test_integrand_prefactor_same_with_numexpr_as_without _______

[gw0] darwin -- Python 3.9.21

self = <holopy.scattering.tests.test_lens.TestLens testMethod=test_integrand_prefactor_same_with_numexpr_as_without>

    @pytest.mark.skipif(not lens.NUMEXPR_INSTALLED,
                        reason = "numexpr package required")
    def test_integrand_prefactor_same_with_numexpr_as_without(self):
        np.random.seed(1649)
        krho, phi, kz = np.random.randn(3, 101)
    
        prefactor_numexpr = LENSMIE._integrand_prefactor(krho, phi, kz)
        prefactor_numpy = LENSMIE_NO_NE._integrand_prefactor(krho, phi, kz)
        if ne.get_vml_version() is None:
>           assert_equal(prefactor_numexpr, prefactor_numpy)
E           AssertionError: 
E           Arrays are not equal
E           
E           Mismatched elements: 7133 / 1010000 (0.706%)
E           Max absolute difference: 3.0304372e-20
E           Max relative difference: 4.57318647e-16
E            x: array([[[5.257655e-10-3.105010e-14j, 5.257655e-10-1.243220e-15j,
E                    5.257655e-10-5.002021e-14j, ..., 5.257655e-10+5.604645e-14j,
E                    5.257655e-10+1.937360e-14j, 5.257655e-10-1.434728e-14j],...
E            y: array([[[5.257655e-10-3.105010e-14j, 5.257655e-10-1.243220e-15j,
E                    5.257655e-10-5.002021e-14j, ..., 5.257655e-10+5.604645e-14j,
E                    5.257655e-10+1.937360e-14j, 5.257655e-10-1.434728e-14j],...

holopy/scattering/tests/test_lens.py:188: AssertionError
========================== short test summary info ===========================
FAILED holopy/scattering/tests/test_lens.py::TestLens::test_integrand_prefactor_same_with_numexpr_as_without - AssertionError: 
=========== 1 failed, 649 passed, 12 skipped, 2 xfailed in 23.62s ============
```

--- a/holopy/scattering/tests/test_lens.py
+++ b/holopy/scattering/tests/test_lens.py
@@ -184,19 +184,20 @@ class TestLens(unittest.TestCase):
 
         prefactor_numexpr = LENSMIE._integrand_prefactor(krho, phi, kz)
         prefactor_numpy = LENSMIE_NO_NE._integrand_prefactor(krho, phi, kz)
-        if ne.get_vml_version() is None:
-            assert_equal(prefactor_numexpr, prefactor_numpy)
-        else:
+        if ne.get_vml_version() is not None:
             ne.set_vml_accuracy_mode('high')
-            # arrays should agree to within a few ULPs. According to Intel, in
-            # high accuracy mode the max error for an elementary function is 1
-            # ULP (4 ULP for low accuracy). But this error is propagated in the
-            # calculation of the prefactor, which involves multiple elementary
-            # function calculations, so we have to set a higher value for nulp.
-            # Minimum value of nulp required for test to pass is 4, based on
-            # experiments on Windows/MKL 20230613)
-            assert_array_almost_equal_nulp(prefactor_numpy, prefactor_numexpr,
-                                           nulp=5)
+
+        # arrays should agree to within a few ULPs. According to
+        # Intel, in high accuracy mode the max error for an elementary
+        # function is 1 ULP (4 ULP for low accuracy). But this error
+        # is propagated in the calculation of the prefactor, which
+        # involves multiple elementary function calculations, so we
+        # have to set a higher value for nulp.  Minimum value of nulp
+        # required for test to pass is 4, based on experiments on
+        # Windows/MKL 20230613). Apple Accelerate BLAS/LAPACK may have
+        # similar accuracy, so we'll set nulp=5 for all platforms
+        assert_array_almost_equal_nulp(prefactor_numpy, prefactor_numexpr,
+                                       nulp=5)
 
     # for the following two tests, numexpr should agree exactly with numpy even
     # if MKL is used; this is because the numexpr expressions involve only

